### PR TITLE
Policy marshalling

### PR DESCRIPTION
**Pull request by francoispradeau**
Add code to handle marshalling/unmarshalling for Istio policies and meshpolicies (taken from https://github.com/michaelkipper/istio-client-go/commit/7c8e95b5d9220d47c107bb6f3b0b71fbc8af3ef7#diff-1c6fa6bfc320013a249f4b6d0ccdd928R65)

### CODE REVIEW

File: pkg/apis/authentication/v1alpha1/policy.go [hunk 1]
```diff
@@ -51,8 +56,34 @@ type PolicySpec struct {
...
+		return err
+	}
+	return nil
+}
+
 // DeepCopyInto is a deepcopy function, copying the receiver, writing into out. in must be non-nil.
 // Based of https://github.com/istio/istio/blob/release-0.8/pilot/pkg/config/kube/crd/types.go#L450
-func (in *PolicySpec) DeepCopyInto(out *PolicySpec) {
-	*out = *in
+func (p *PolicySpec) DeepCopyInto(out *PolicySpec) {
```
Comment (nrjpoddar) [inline pkg/apis/authentication/v1alpha1/policy.go]: *in and *out is a generic name for DeepCopy functions, I would suggest keeping it the same.

File: Gopkg.toml [hunk 2]
```diff
@@ -67,3 +67,7 @@ required = [
     unused-packages = false
     non-go = false
     go-tests = false
+
+[[constraint]]
```
Comment (nrjpoddar) [inline Gopkg.toml]: Did you add this manually or dep upated Gopkg.toml?

File: pkg/apis/authentication/v1alpha1/policy.go (hunk 1 above)
Comment (francoispradeau) [reply pkg/apis/authentication/v1alpha1/policy.go]: Didn't think I changed that... will revert to original.

File: Gopkg.toml (hunk 2 above)
Comment (francoispradeau) [reply Gopkg.toml]: dep updated Gopkg.toml...

**Pull request merged**